src/main/java/org/distributed/systems/chord/messaging/Predecessor.java

```diff
@@ -12,12 +10,10 @@ public Request() {
     }
 
     public static class Reply implements Response {
-        public ActorRef predecessor;
-        public long predecessorId;
+        public ChordNode predecessor;
 
-        public Reply(ActorRef predecessor, long predecessorId) {
```

Comment:
Perfect! I found my implementation quite annoying, and wanted to encapsulated that too! :+1: